### ISSUE TITLE
GregTech 5 Unofficial power conversion integration?

### ISSUE THREAD

**Issue by Barhandar** [ReikaKalseki/Reika_Mods_Issues#1590]
Currently, V18b, RotaryCraft cannot directly convert to and from GT5U-EU (which shares units with IC2's EU, but is more consistent and behaves slightly differently). It would be nice if Electric Motor/Generator could do that.

>RotaryCraft V18b, IC2E 2.2.827, GT5U 5.09.30.  
>Electric Generator does not function or has non-obvious requirements. I've tried with Gasoline Engine, Gas Turbine and Creative Coil as input, and MFSU/BatBox/Tin Cable into BatBox/Glass Fiber into BatBox, the batteries do not charge and EU Reader on the Generator shows all zeroes. IC2E's cables still connect visually to the output side.  
>Electric Motor only functions when connected to IC2E's sources, and is not recognized by GT5U as valid IC2-EU-conversion target (GT5U will charge IC2E BatBoxes, but it won't charge the Motor). Magnetostatic Engine is recognized by GT5U as valid RF container (by default, GT will output RF to anything that accepts it, but not the reverse), so I think the issue is with the Electric Motor.

**Comment by ReikaKalseki** [comment]
Its API must be incompatible.

**Comment by OvermindDL1** [comment]
> Its API must be incompatible.

Correct.  GT5 (and GT6) can output IC2 energy packets, but it also outputs its own form of energy packets (that do not have certain issues the IC2 versions do), and they only accept its own form of energy packets unless something outputs IC2 power to a GT5/GT6 Transformer (the only block that accepts IC2 power as it has the code to do the proper conversions in a non-exploitable way).

For now though, anyone using GT5/GT6 can just use a transformer to take 'in' IC2 power (anything that implements the IC2 power interface), and they should be able to power anything with GT5/GT6 that accepts IC2 power straight.

If GT5/GT6 cannot power Roterycraft things that accept IC2 then it is missing something about the IC2 interface on the block.

**Comment by Barhandar** [comment]
It's a different API from IC2's, [but it's still possible to, er, interface with it with some crowbaring](https://github.com/BluSunrize/ImmersiveEngineering/blob/1.7.10/src/main/java/blusunrize/immersiveengineering/common/util/compat/GregTechHelper.java#L138).

**Comment by OvermindDL1** [comment]
> It's a different API from IC2's, but it's still possible to, er, interface with it with some crowbaring.

Why would that even be needed?  Just implement the IC2 interface and it'll work with GT too (just without the more advanced features).

**Comment by ReikaKalseki** [comment]
I *do* implement the IC2 interfaces...

**Comment by Barhandar** [comment]
>Why would that even be needed? Just implement the IC2 interface and it'll work with GT too (just without the more advanced features).

The point is to be able to power GT machines directly like IE does, instead of requiring a transformer.
>I do implement the IC2 interfaces...

But for some reason they only function for input (EU-to-Shaft) with IC2E, and don't function at all with GT5U, when GT5U is installed.
